### PR TITLE
Added a flag to disable shadow casting on the core material types.

### PR DESCRIPTION
**Pull request by santorac**
## What does this PR do?

Added a flag to disable shadow casting on the core material types.
This addresses https://github.com/o3de/o3de/issues/12897.

I also added the capability for material properties to connect directly to the Enable state of a shader, similar to a direct shader constant or shader option connection. I ended up only being able to use this in one material type, but I think it will become more useful in the future.

Fixed ShadowCatcher shader (https://github.com/o3de/o3de/issues/10678). The opacity mode was set to Opaque which caused it to attempt using the fullscreen shadow buffer, but fullscreen shadows aren't supported for transparent materials. I also noticed some functions had an unused screen position parameter, which I removed.

Signed-off-by: santorac <55155825+santorac@users.noreply.github.com>

## How was this PR tested?

AtomSampleViewer full test suite on vulkan saw no changes.
Tested each material type in Material Editor, flipping both the Receive Shadows and Cast Shadows flags. These affect self-shadowing and the shader catcher plane as expected.

### CODE REVIEW

**Review by antonmic — APPROVED** [review]

File: Gems/Atom/Feature/Common/Assets/Materials/Special/ShadowCatcher.azsl [hunk 1]
```diff
@@ -15,7 +15,7 @@ option enum class OpacityMode
     Blended, 
     TintedTransparent
     
-} o_opacity_mode = OpacityMode::Opaque;
+} o_opacity_mode = OpacityMode::Blended;
```
Comment (rgba16f) [inline Gems/Atom/Feature/Common/Assets/Materials/Special/ShadowCatcher.azsl:18]: is this a debugging change? Doesn't seem related to the shadow enabled change.

**Review by rgba16f — APPROVED** [review]

**Review by gadams3 — APPROVED** [review]

**Comment by ParasiteSt** [comment]
Is there a required PR in front of this one before it can be merged?

File: Gems/Atom/Feature/Common/Assets/Materials/Special/ShadowCatcher.azsl (hunk 1 above)
Comment (santorac) [reply Gems/Atom/Feature/Common/Assets/Materials/Special/ShadowCatcher.azsl:18]: This is "Fixed ShadowCatcher shader" mentioned in the description.

**Comment by santorac** [comment]
> Is there a required PR in front of this one before it can be merged?

No, it's ready to go, we just need to get passed Automated Review.
